lib/Service/CurlService.php

```diff
@@ -68,6 +76,83 @@ public function __construct(ConfigService $configService, MiscService $miscServi
 	}
 
 
+	/**
+	 * @param string $account
+	 *
+	 * @return Person
+	 * @throws InvalidResourceException
+	 * @throws MalformedArrayException
+	 * @throws Request410Exception
+	 * @throws RequestException
+	 * @throws SocialAppConfigException
+	 * @throws RedundancyLimitException
+	 * @throws UnknownItemException
+	 * @throws InvalidOriginException
+	 */
+	public function retrieveAccount(string $account): Person {
+		$account = $this->withoutBeginAt($account);
+
+		if (strstr(substr($account, 0, -3), '@') === false) {
```

Comment:
This check doesn't seem right `@test` will also be a valid account then. Having the additional null check after the explode would be fine I think.